public/app/features/alerting/unified/components/rules/central-state-history/EventListSceneObject.tsx

```diff
@@ -395,24 +477,67 @@ export class HistoryEventsListObject extends SceneObjectBase {
   }
 }
 
+export type FilterType = 'label' | 'stateFrom' | 'stateTo';
+
 export function HistoryEventsListObjectRenderer({ model }: SceneComponentProps<HistoryEventsListObject>) {
   const { value: timeRange } = sceneGraph.getTimeRange(model).useState(); // get time range from scene graph
-  const filtersVariable = sceneGraph.lookupVariable(LABELS_FILTER, model)!;
-
-  const valueInfilterTextBox: VariableValue = !(filtersVariable instanceof TextBoxVariable)
-    ? ''
-    : filtersVariable.getValue();
+  // eslint-disable-next-line
+  const labelsFiltersVariable = sceneGraph.lookupVariable(LABELS_FILTER, model)! as TextBoxVariable;
+  // eslint-disable-next-line
+  const stateToFilterVariable = sceneGraph.lookupVariable(STATE_FILTER_TO, model)! as CustomVariable;
+  // eslint-disable-next-line
+  const stateFromFilterVariable = sceneGraph.lookupVariable(STATE_FILTER_FROM, model)! as CustomVariable;
+
+  const valueInfilterTextBox: VariableValue = labelsFiltersVariable.getValue();
+  const valueInStateToFilter = stateToFilterVariable.getValue();
+  const valueInStateFromFilter = stateFromFilterVariable.getValue();
+
+  const addFilter = (key: string, value: string, type: FilterType) => {
+    const newFilterToAdd = `${key}=${value}`;
+    if (type === 'stateTo') {
+      stateToFilterVariable.changeValueTo(value);
+    }
+    if (type === 'stateFrom') {
+      stateFromFilterVariable.changeValueTo(value);
+    }
+    if (type === 'label') {
+      const finalFilter = valueInfilterTextBox
+        ? `${valueInfilterTextBox.toString()} ,${newFilterToAdd}`
```

Comment:
nit: 
```suggestion
        ? `${valueInfilterTextBox.toString()}, ${newFilterToAdd}`
```
(Unless this breaks some edge case? I couldn't see a problem with doing this when I tried it out)